upgrade/scripts/upgrade/prerequisites.sh

```diff
@@ -1263,6 +1263,55 @@ else
   echo "====> ${state_name} has been completed" | tee -a "${LOG_FILE}"
 fi
 
+state_name="PREPARE_KUBEADM"
+state_recorded=$(is_state_recorded "${state_name}" "$(hostname)")
+if [[ ${state_recorded} == "0" ]]; then
+  echo "====> ${state_name} ..." | tee -a "${LOG_FILE}"
+  {
+    tmpdir=$(mktemp -d)
+
+    echo "Patching ConfigMap kubeadm-config ..."
+    kubectl -n kube-system get configmap kubeadm-config -o go-template --template '{{ .data.ClusterConfiguration }}' \
+      | yq4 e '.kubernetesVersion="1.24.17"' \
+      | yq4 e '.dns.imageRepository="artifactory.algol60.net/csm-docker/stable/registry.k8s.io/coredns"' \
+      | yq4 e '.imageRepository="artifactory.algol60.net/csm-docker/stable/registry.k8s.io"' \
```

Comment:
I'm not sure if this is the right place to fix this in the upgrade, but could you also add:
```
    | yq4 e '.controllerManager.extraArgs.terminated-pod-gc-threshold="250"' \
```
See: https://jira-pro.it.hpe.com:8443/browse/CASMSEC-411

Reply:
This is the right place for CSM 1.5 > 1.6 upgrade flow. I have applied the change. For fresh install of CSM 1.6 onwards, the right place would be here: https://github.com/Cray-HPE/node-images/blob/main/boxes/ncn-node-images/kubernetes/files/scripts/metal/lib.sh#L42

Reply:
Great!  And thanks for updating the CASMSEC-411 ticket!

Reply:
Since we're making relevant changes here, could you also add:
`| yq4 e '.controllerManager.extraArgs.profiling="false"' \`

See: [CASM-4576](https://jira-pro.it.hpe.com:8443/browse/CASM-4576)

Reply:
Done!